### PR TITLE
add nf-test and fix stub happy/sompy

### PR DESCRIPTION
**Pull request by kubranarci**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/happy/sompy/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.stats).match("stats_1") },
+                { assert snapshot(process.out.versions).match("versions_1") },
+                { assert snapshot(file(process.out.metrics[0][1]).name).match("metrics_1")},
+                { assert snapshot(file(process.out.features[0][1]).name).match("features_1")}
```
Comment (SPPearce) [inline modules/nf-core/happy/sompy/tests/main.nf.test:45]: Can you put all the snapshots into one: (indentation probably needs fixing)
```suggestion
                { assert snapshot(
                  process.out.stats,
                  process.out.versions,
                  file(process.out.metrics[0][1]).name,
                  file(process.out.features[0][1]).name
                ).match() },
```

Comment (kubranarci) [reply modules/nf-core/happy/sompy/tests/main.nf.test:45]: when I don't separate "versions", lint gives an error. that is why I put it like this.

Comment (SPPearce) [reply modules/nf-core/happy/sompy/tests/main.nf.test:45]: Hmm, shouldn't be an issue like this, I've done it on a number of modules

Comment (kubranarci) [reply modules/nf-core/happy/sompy/tests/main.nf.test]: I dont know what was that, but it works out inside one match now

**Review by SPPearce — APPROVED** [review]

**Pull request merged**